src/service/BlockService.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2020 NEM
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { sha3_256 } from 'js-sha3';
+import { combineLatest, Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { BlockRepository } from '../infrastructure/BlockRepository';
+import { ReceiptRepository } from '../infrastructure/ReceiptRepository';
+import { MerklePathItem } from '../model/blockchain/MerklePathItem';
+import { UInt64 } from '../model/UInt64';
+
+/**
+ * Transaction Service
+ */
+export class BlockService {
+
+    /**
+     * Constructor
+     * @param blockRepository
+     * @param receiptRepository
+     */
+    constructor(private readonly blockRepository: BlockRepository, private readonly receiptRepository: ReceiptRepository) {
+    }
+
+    /**
+     * Validate transaction hash in block
+     * @param leaf transaction hash
+     * @param height block height
+     */
+    public validateTransactionInBlock(leaf: string, height: UInt64): Observable<boolean> {
+        const rootHashObservable = this.blockRepository.getBlockByHeight(height);
+        const merklePathItemObservable = this.blockRepository.getMerkleTransaction(height, leaf);
+        return combineLatest(rootHashObservable, merklePathItemObservable).pipe(
+            map((combined) => this.validateInBlock(leaf, combined[1].merklePath, combined[0].blockTransactionsHash)),
+        );
+    }
+
+    /**
+     * Validate receipt hash in block
+     * @param leaf receipt hash
+     * @param height block height
+     */
+    public validateReceiptInBlock(leaf: string, height: UInt64): Observable<boolean> {
```

Comment:
Should the function be named ``validateStatementInBlock``? The hash passed as leaf belongs to a transaction/resolution statement, not to an individual receipt.